pymeasure/instruments/pendulum/cnt91.py

```diff
@@ -144,45 +145,46 @@ def read_buffer(self, expected_length=0):
         map_values=True,
     )
 
-    def configure_frequency_array_measurement(self, n_samples, channel):
+    def configure_frequency_array_measurement(self, n_samples, channel, back_to_back=True):
         """
         Configure the counter for an array of measurements.
 
         :param n_samples: The number of samples
         :param channel: Measurement channel (A, B, C, E, INTREF)
+        :param back_to_back: If True, the buffer measurement is performed back-to-back.
         """
-        n_samples = truncated_range(n_samples, [1, self.MAX_BUFFER_SIZE])
+        n_samples = truncated_range(n_samples, [MIN_BUFFER_SIZE, MAX_BUFFER_SIZE])
         channel = strict_discrete_set(channel, self.CHANNELS)
         channel = self.CHANNELS[channel]
-        self.write(f":CONF:ARR:FREQ {n_samples},(@{channel})")
+        self.write(f":CONF:ARR:FREQ{':BTB' if back_to_back else ''} {n_samples},(@{channel})")
 
     def buffer_frequency_time_series(
-        self, channel, n_samples, sample_rate, trigger_source=None
+        self,
+        channel,
+        n_samples,
+        *,
+        gate_time,
+        trigger_source=None,
+        back_to_back=True,
+        sample_rate=None,
     ):
         """
         Record a time series to the buffer and read it out after completion.
 
         :param channel: Channel that should be used
         :param n_samples: The number of samples
-        :param sample_rate: Sample rate in Hz
-        :param trigger_source: Optionally specify a trigger source to start the
-            measurement
+        :param gate_time: Gate time in s
+        :param trigger_source: Optionally specify a trigger source to start the measurement
+        :param back_to_back: If True, the buffer measurement is performed back-to-back.
```

Comment:
You should keep the deprecated parameter in the list, but with a deprecation notice:
```suggestion
        :param back_to_back: If True, the buffer measurement is performed back-to-back.
        :param sample_rate: Sample rate in Hz

            .. deprecated:: 0.14
                Use parameter `gate_time` instead.
```